app/src/main/java/com/eveningoutpost/dexdrip/insulin/LinearTrapezoidInsulin.java

```diff
@@ -37,18 +37,18 @@ public double calculateIOB(double t) {
         else if ((t1 <= time) && (time < t2))
             return 1.0 + 0.5 * max * t1 - max * time;
         else if ((t2 <= time) && (time < t3))
-            return 0.5 * (time - t3) * (time - t3) * max / (t3 - t2);
+            return 0.5 * (t3 - time) * (t3 - time) * max / (t3 - t2);
         else return 0;
     }
 
     public double calculateActivity(double t) {
         double time = t - onset;
         if ((0 <= time) && (time < t1))
-            return concentration * time * max / t2;
+            return concentration * time * max / t1;
```

Comment:
thats already fixed in Features/MultipleInsulin (but not in this branch). Thanks for finding and fixing